server/src/main/java/io/druid/server/lookup/cache/LookupCoordinatorManager.java

```diff
@@ -506,137 +257,398 @@ public boolean deleteLookup(final String tier, final String lookup, AuditInfo au
+        lifecycleLock.exitStop();
+        lifecycleLock.reset();
+      }
+    }
+  }
+
+  private void initializeLookupsConfigWatcher()
+  {
+    //Note: this call is idempotent, so multiple start() would not cause any problems.
+    lookupMapConfigRef = configManager.watch(
+        LOOKUP_CONFIG_KEY,
+        new TypeReference<Map<String, Map<String, LookupExtractorFactoryMapContainer>>>()
         {
-          backgroundManagerExitedLatch.countDown();
-          if (backgroundManagerFuture.isCancelled()) {
-            LOG.info("Background lookup manager exited");
-            LOG.trace(t, "Background lookup manager exited with throwable");
-          } else {
-            LOG.makeAlert(t, "Background lookup manager exited with error!").emit();
+        },
+        null
+    );
+
+    // backward compatibility with 0.10.0
+    if (lookupMapConfigRef.get() == null) {
+      Map<String, Map<String, Map<String, Object>>> oldLookups = configManager.watch(
+          OLD_LOOKUP_CONFIG_KEY,
+          new TypeReference<Map<String, Map<String, Map<String, Object>>>>()
+          {
+          },
+          null
+      ).get();
+
+      if (oldLookups != null) {
+        Map<String, Map<String, LookupExtractorFactoryMapContainer>> converted = new HashMap<>();
+        for (Map.Entry<String, Map<String, Map<String, Object>>> tierEntry : oldLookups.entrySet()) {
+          Map<String, Map<String, Object>> oldTierLookups = tierEntry.getValue();
+          if (oldTierLookups != null && !oldTierLookups.isEmpty()) {
+            Map<String, LookupExtractorFactoryMapContainer> convertedTierLookups = new HashMap<>();
+            for (Map.Entry<String, Map<String, Object>> e : oldTierLookups.entrySet()) {
+              convertedTierLookups.put(e.getKey(), new LookupExtractorFactoryMapContainer(null, e.getValue()));
+            }
+            converted.put(tierEntry.getKey(), convertedTierLookups);
           }
         }
-      });
-      started = true;
-      LOG.debug("Started");
+        configManager.set(
+            LOOKUP_CONFIG_KEY,
+            converted,
+            new AuditInfo("autoConversion", "autoConversion", "127.0.0.1")
+        );
+      }
     }
   }
 
-  @LifecycleStop
-  public void stop()
+  private void lookupManagementLoop()
   {
-    synchronized (startStopSync) {
-      if (!started) {
-        LOG.warn("Not started, ignoring stop request");
-        return;
+    // Sanity check for if we are shutting down
+    if (Thread.currentThread().isInterrupted() || !lifecycleLock.awaitStarted(1, TimeUnit.MILLISECONDS)) {
+      LOG.info("Not updating lookups because process was interrupted or not finished starting yet.");
+      return;
+    }
+
+    final Map<String, Map<String, LookupExtractorFactoryMapContainer>> allLookupTiers = lookupMapConfigRef.get();
+
+    if (allLookupTiers == null) {
+
+      LOG.debug("Not updating lookups because no data exists");
```

Comment:
Since `lookupManagementLoop()` is called rarely, I think it makes sense to log it on INFO level, i. e. "always on". It helps to debug problems after-the-fact without turing on all debug output. Recently we had such situation in autoscaling, and logging helped to find a bug.